kotlin-grpc/src/test/kotlin/CoroutineServerInterceptorTest.kt

```diff
@@ -0,0 +1,101 @@
+import io.grpc.Metadata
+import io.grpc.ServerCall
+import io.grpc.ServerCallHandler
+import io.grpc.Status
+import org.junit.jupiter.api.extension.RegisterExtension
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.TimeUnit
+import kotlin.coroutines.suspendCoroutine
+
+internal class CoroutineServerInterceptorTest {
+    private class AuthInterceptor : CoroutineServerInterceptor {
+        private val authorizer = Authorizer { ctx: ServiceRequestContext, metadata: Metadata? ->
+            val future = CompletableFuture<Boolean>()
+            ctx.eventLoop().schedule({
+                if (ctx.request().headers().contains("Authorization", "Bearer token-1234")) {
+                    future.complete(true)
+                } else {
+                    future.complete(false)
+                }
+            }, 100, TimeUnit.MILLISECONDS)
+            return@Authorizer future
+        }
+
+        override suspend fun <ReqT, RespT> suspendedInterceptCall(
+            call: ServerCall<ReqT, RespT>,
+            headers: Metadata,
+            next: ServerCallHandler<ReqT, RespT>
+        ): ServerCall.Listener<ReqT> = suspendCoroutine {
+            val future = authorizer.authorize(ServiceRequestContext.current(), headers)
+            future.whenComplete { result, _ ->
+                if (result) {
+                    next.startCall(call, headers)
+                } else {
+                    throw AnticipatedException("Invalid access")
+                }
+            }
+        }
+    }
+
+    companion object {
+        @RegisterExtension
+        val server: ServerExtension = object : ServerExtension() {
+            override fun configure(sb: ServerBuilder) {
+                val statusFunction = GrpcStatusFunction { ctx: RequestContext?, throwable: Throwable, metadata: Metadata? ->
+                    if (throwable is AnticipatedException && "Invalid access" == throwable.message) {
+                        return@GrpcStatusFunction Status.UNAUTHENTICATED
+                    }
+                    null
+                }
+                val authInterceptor = AuthInterceptor()
+                sb.serviceUnder(
+                    "/non-blocking",
+                    GrpcService.builder()
+                        .exceptionMapping(statusFunction)
+                        .intercept(authInterceptor)
+                        .build()
+                )
+                sb.serviceUnder(
+                    "/blocking",
+                    GrpcService.builder()
+                        .exceptionMapping(statusFunction)
+                        .intercept(authInterceptor)
+                        .useBlockingTaskExecutor(true)
+                        .build()
+                )
+            }
+        }
+
+        object TestServiceImpl {
+            fun todo() {
+                TODO("If we need same test codes like AsyncServerInterceptorTest, need TestServiceImpl in grpc module's test dir")
```

Comment:
Are we going to add tests? 😆

Reply:
I tried to write using the same test code as the previously written `AsyncServerInterceptorTest`, but using `AuthInterceptor` implementing `CoroutineServerInterceptor`.

However, `TestServiceImpl`, the class used by `AsyncServerInterceptorTest`, is under `test` dir, so it cannot be used to implement the `grpc` module 😓 

Either copy the code or the `AsyncServerInterceptor` has been sufficiently verified with the test code, so it may not be necessary to have the test code at this stage. What do you think? 🤔

Reply:
> However, TestServiceImpl, the class used by AsyncServerInterceptorTest, is under test dir, so it cannot be used to implement the grpc module 😓

We can copy the `test.proto` file in the grpc module or we can define a new service for the test. Could you add one, please?

 > Either copy the code or the AsyncServerInterceptor has been sufficiently verified with the test code, so it may not be necessary to have the test code at this stage.
 
We use coroutine and `AsyncServerInterceptorTest` uses future inside. They are not equivalent so I don't think we can substitute for the test. 😉

Reply:
So you mean we need some test cases for new Interceptor...
I'll try 🙂

Reply:
In [47cdd88](https://github.com/line/armeria/pull/4724/commits/47cdd8809fc59932dc6a51a053bc26ddfddab642), I added some test cases as same as [AsyncServerInterceptorTest](https://github.com/line/armeria/blob/master/grpc/src/test/java/com/linecorp/armeria/server/grpc/AsyncServerInterceptorTest.java).